src/logger/logger.py

```diff
@@ -21,470 +24,521 @@
 
 
 # pylint: disable=too-many-public-methods
-class Logger:
+class Logger(Component):
     """
     The logger class is used to log the events of the simulation
     """
 
     run_id: UUID
+    callback_handles: list[UUID]
 
-    def __init__(self, run_id: UUID):
+    def __init__(self, run_id: UUID, event_bus: EventBus):
         """
         The constructor of the logger class
         """
+        super().__init__(event_bus, 5)
         self.run_id = run_id
```

Comment:
Does the Logger still need run_id for anything or could it be removed? I was wondering, because the EventBus also has run_id.

Reply:
You are right. I removed it. Thanks :)